corgi/core/migrations/0040_auto_20221020_1057.py

```diff
@@ -0,0 +1,47 @@
+# Generated by Django 3.2.15 on 2022-10-20 10:57
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("core", "0039_auto_20221013_0748"),
+    ]
+
+    operations = [
+        migrations.AddIndex(
+            model_name="product",
+            index=models.Index(fields=["ofuri"], name="core_produc_ofuri_671dbe_idx"),
+        ),
+        migrations.AddIndex(
+            model_name="productstream",
+            index=models.Index(fields=["ofuri"], name="core_produc_ofuri_e77387_idx"),
+        ),
+        migrations.AddIndex(
+            model_name="productvariant",
+            index=models.Index(fields=["ofuri"], name="core_produc_ofuri_ccf4bd_idx"),
+        ),
+        migrations.AddIndex(
+            model_name="productversion",
+            index=models.Index(fields=["ofuri"], name="core_produc_ofuri_c62cf2_idx"),
+        ),
+        migrations.RunSQL(
+            "CREATE INDEX core_compon_latest_name_type_idx ON public.core_component \
+             USING btree (name, type) WHERE (((type)::text = 'SRPM'::text) OR \
+             (((arch)::text = 'noarch'::text) AND ((type)::text = 'CONTAINER_IMAGE'::text)) \
+             OR ((type)::text = 'RHEL_MODULE'::text))"
+        ),
+        migrations.RunSQL(
```

Comment:
Why is there more than one index being added here?

Reply:
to aid subquery ... note that these partial queries are very cheap in terms of storage and ingestion performance eg. around 12MB ... the bigger the table the more useful having foward/backward type queries